keanu-project/src/main/java/io/improbable/keanu/tensor/Tensor.java

```diff
@@ -119,10 +119,14 @@ default T getValue(int... index) {
         return Arrays.asList(asFlatArray());
     }
 
-    default boolean isScalar() {
+    default boolean isLengthOne(){
```

Comment:
Singular already has a mathematical meaning. `IsLengthOne` was the clearest and shortest name I could come up with. Really any code that depends on this smells because it discounts the shape any only cares if it's length 1. For example, we multiply by a scalar when multiplying anything that is length 1. This could cause the result to be the wrong shape. It's out of scope for this PR so I left it but I'll add a ticket to look at it.